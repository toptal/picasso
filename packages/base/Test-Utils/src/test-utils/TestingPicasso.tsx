import React from 'react'
import Picasso from '@toptal/picasso-provider'
import type { TextLabelProps } from '@toptal/picasso-shared'

export type Props = TextLabelProps & {
  children: React.ReactNode
  injectFirst?: boolean
}

export const TestingPicasso = ({
  children,
  titleCase,
  injectFirst = false,
}: Props) => {
  return (
    <Picasso
      loadFavicon={false}
      loadFonts={false}
      fixViewport={false}
      preventPageWidthChangeOnScrollbar={false}
      titleCase={titleCase}
      disableTransitions
      injectFirst={injectFirst}
    >
      {children}
    </Picasso>
  )
}

TestingPicasso.displayName = 'TestingPicasso'

export default TestingPicasso
