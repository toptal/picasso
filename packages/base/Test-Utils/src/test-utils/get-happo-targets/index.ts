import { getCheckpoints } from './get-checkpoints'
import { getHappoTargets } from './get-happo-targets'

const checkpoints = getCheckpoints()

export { getHappoTargets, getCheckpoints }

export const HAPPO_TARGETS = getHappoTargets(checkpoints)
