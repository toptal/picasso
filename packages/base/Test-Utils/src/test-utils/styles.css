/* we can't apply reset.css here as it would override current styles */
/* @tailwind base; */

@tailwind components;
@tailwind utilities;
