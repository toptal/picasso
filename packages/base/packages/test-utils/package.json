{
  "name": "@toptal/picasso-test-utils",
  "version": "1.0.0",
  "main": "index.js",
  "dependencies": {}
}