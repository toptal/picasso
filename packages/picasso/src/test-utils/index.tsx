import type { ReactElement } from 'react'
import React from 'react'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'

import TestingPicasso from './TestingPicasso'

export type PicassoConfig = {
  titleCase: boolean
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  picassoConfig?: PicassoConfig
): RenderResult => {
  const { rerender, ...rest } = render(
    <TestingPicasso titleCase={picassoConfig?.titleCase}>{ui}</TestingPicasso>,
    options
  )

  const customRerender: typeof rerender = newUi =>
    rerender(
      <TestingPicasso titleCase={picassoConfig?.titleCase}>
        {newUi}
      </TestingPicasso>
    )

  return { rerender: customRerender, ...rest }
}

export * from '@testing-library/react'
export { customRender as render, TestingPicasso }
export { getCheckpoints } from './get-checkpoints'
