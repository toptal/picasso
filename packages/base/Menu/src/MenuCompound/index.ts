import { Menu } from '../Menu'
import { MenuItem } from '../MenuItem'

type MenuCompoundType = typeof Menu & {
  Item: typeof MenuItem
}

export const MenuCompound: MenuCompoundType = Object.assign(Menu, {
  Item: MenuItem,
})
