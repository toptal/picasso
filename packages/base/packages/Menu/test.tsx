import React from 'react'
import { render } from '@toptal/picasso/test-utils'

import { MenuCompound as Menu } from '../MenuCompound'

describe('Menu', () => {
  it('renders', () => {
    const { container } = render(
      <Menu>
        <Menu.Item>1</Menu.Item>
        <Menu.Item>2</Menu.Item>
      </Menu>
    )

    expect(container).toMatchSnapshot()
  })
})
