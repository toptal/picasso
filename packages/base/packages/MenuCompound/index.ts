/* eslint-disable import/no-extraneous-dependencies */
import Menu from '@toptal/picasso-menu'
import MenuItem from '@toptal/picasso-menu-item'

export const MenuCompound = Object.assign(Menu, { Item: MenuItem })
