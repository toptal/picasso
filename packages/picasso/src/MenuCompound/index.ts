import Menu from '../Menu'
import MenuItem from '../MenuItem'

export const MenuCompound = Object.assign(Menu, { Item: MenuItem })
