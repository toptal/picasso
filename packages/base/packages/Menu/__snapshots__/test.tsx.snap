// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <ul
      class="MuiList-root PicassoMenu-root"
      role="menu"
      tabindex="-1"
    >
      <li
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root PicassoMenuItem-root PicassoMenuItem-light MuiMenuItem-gutters PicassoMenuItem-gutters MuiListItem-gutters MuiListItem-button"
        role="menuitem"
        tabindex="0"
      >
        <div
          class="PicassoContainer-flex PicassoMenuItem-itemWrapper"
        >
          <div
            class="PicassoContainer-flex PicassoContainer-column PicassoMenuItem-content"
          >
            <div
              class="PicassoContainer-centerAlignItems PicassoContainer-flex"
            >
              <span
                class="PicassoMenuItem-stringContent"
              >
                1
              </span>
            </div>
          </div>
        </div>
      </li>
      <li
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root PicassoMenuItem-root PicassoMenuItem-light MuiMenuItem-gutters PicassoMenuItem-gutters MuiListItem-gutters MuiListItem-button"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="PicassoContainer-flex PicassoMenuItem-itemWrapper"
        >
          <div
            class="PicassoContainer-flex PicassoContainer-column PicassoMenuItem-content"
          >
            <div
              class="PicassoContainer-centerAlignItems PicassoContainer-flex"
            >
              <span
                class="PicassoMenuItem-stringContent"
              >
                2
              </span>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
